src/sbvr-api/control-flow.ts

```diff
@@ -0,0 +1,61 @@
+import * as _ from 'lodash'
+import * as Promise from 'bluebird'
+
+// lifts f to map over the changeSet if present
+export const liftP = <T, U>(fn: (v: T) => U | Promise<U>) => {
+	return ((a: T | T[]) => {
+		if (_.isArray(a)) {
+			// This must not be a settle as if any operation fails in a cs
+			// we want to discard the whole
+			return Promise.mapSeries(a, fn)
+		} else {
+			return Promise.resolve(a).then(fn)
+		}
+	// TODO: We cast here to merge `Promise<U> | Promise<U[]>` into `Promise<U | U[]>,
+	// which is equivalent for our uses but fixes issues where typing was being rejected
+	}) as ((a: T | T[]) => Promise<U | U[]>)
```

Comment:
Instead of casting, can you try specifying the function's return type explicitly and see if that works?

Also, why is this not a proper function? Function variables suck because they're anonymous and therefore invisible in back traces.